SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -45,6 +55,128 @@ public MainWindow()
+            SetMarkers(sender, e);
+        }
+        private void MapMouseLeftButtonDown(object sender, MouseButtonEventArgs e)
+        {
+            //Double tap on a map
+            if (e.ClickCount == 2 && GridSellerAdd.Visibility == Visibility.Visible)
+            {
+                userSelectedLocation = true;
+                MainMap.TargetCenter = MainMap.ViewToLocation(e.GetPosition(MainMap));
+                SYLMap.AddMarker(MainMap.TargetCenter, "Selected location");
+            }
+            else
+            {
+                userSelectedLocation = false;
+                SYLMap.RemoveLastMarker();
+                if (updateMarketplacesWasClicked)
+                    SetMarkers(sender, e);
+            }
+        }
+
+        private void LabelAddSeller_Click(object sender, RoutedEventArgs e)
+        {
+            //Clean up
+            ErrorLabel1.Visibility = Visibility.Collapsed;
+            TextProduct.Clear();
+            TextTime.Clear();
+            GridSellerAdd.Visibility = Visibility.Visible;
+        }
+
+        private void ButtonSave_Clicked(object sender, RoutedEventArgs e)
+        {
+            // Saving data to csv file
+            if (userSelectedLocation)
+            {
+                SaveData();
+                GridSellerAdd.Visibility = Visibility.Collapsed;//------------------------------------------------------------------
+                SYLMap.RemoveLastMarker();
+                if (updateMarketplacesWasClicked)
+                    SetMarkers(sender, e);
+            }
+            else
+            {
+                ErrorLabel1.Visibility = Visibility.Visible;
+            }
+        }
+
+        private void SaveData()
+        {
+            // Checking person's ID
+            List<double> personID = new List<double>();
+            using (TextFieldParser csvParser = new TextFieldParser(filePath))
+            {
+                csvParser.CommentTokens = new string[] { "#" };
+                csvParser.SetDelimiters(new string[] { "," });
+                csvParser.HasFieldsEnclosedInQuotes = true;
+
+                // Skip the row with the column names
+                csvParser.ReadLine();
+
+                while (!csvParser.EndOfData)
+                {
+                    // Read current line fields, pointer moves to the next line.
+                    string[] fields = csvParser.ReadFields();
+                    personID.Add(Double.Parse(fields[4]));
+                }
+            }
+
+            String product = TextProduct.Text;
+            String time = TextTime.Text;
+            var csv = new StringBuilder();
+
+
+            // Setting person's ID
+            personsID = personsID + personID.Count;
+
+            var newLine = string.Format("{0},{1},{2},{3}", product, time, MainMap.TargetCenter, personsID);
+
+            csv.AppendLine(newLine);
+
+            File.AppendAllText(filePath, csv.ToString());
+        }
```

Comment:
Logic of reading or saving data cannot be inside a WPF code class. It has to be somewhere else, for instance- SupportYourLocals/Services/Csv/ReadingData, but not in SupportYourLocals.WPF/MainWindow.xaml.cs